lib/plugins/aws/provider/awsProvider.js

```diff
@@ -325,6 +330,74 @@ class AwsProvider {
         provider: {
           properties: {
             apiGateway: { type: 'object', properties: { websocketApiId: { type: 'string' } } },
+            alb: {
+              type: 'object',
+              properties: {
+                targetGroupPrefix: { type: 'string', maxLength: 16 },
+                authorizers: {
+                  type: 'object',
+                  additionalProperties: {
+                    type: 'object',
+                    properties: {
+                      type: { enum: ['cognito', 'oidc'] },
+                      userPoolArn: { $ref: '#/definitions/awsArnString' },
+                      userPoolClientId: { type: 'string' },
+                      userPoolDomain: { type: 'string' },
+                      onUnauthenticatedRequest: { enum: ['allow', 'authenticate', 'deny'] },
+                      authorizationEndpoint: { format: 'uri' },
+                      clientId: { type: 'string' },
+                      clientSecret: { type: 'string' },
+                      useExistingClientSecret: { type: 'boolean' },
+                      issuer: { format: 'uri' },
+                      tokenEndpoint: { format: 'uri' },
+                      userInfoEndpoint: { format: 'uri' },
+                      requestExtraParams: {
+                        type: 'object',
+                        maxProperties: 10,
+                        additionalProperties: { type: 'string' },
+                      },
+                      scope: { type: 'string' },
+                      sessionCookieName: { type: 'string' },
+                      sessionTimeout: { type: 'integer' },
+                    },
+                    allOf: [
+                      {
+                        if: {
+                          properties: {
+                            type: { const: 'cognito' },
+                          },
+                        },
+                        then: {
+                          required: ['userPoolArn', 'userPoolClientId', 'userPoolDomain'],
+                        },
+                      },
+                      {
+                        if: {
+                          properties: {
+                            type: { const: 'oidc' },
+                          },
+                        },
+                        then: {
+                          required: [
+                            'authorizationEndpoint',
+                            'clientId',
+                            'issuer',
+                            'tokenEndpoint',
+                            'userInfoEndpoint',
+                          ],
+                          oneOf: [
+                            { required: ['clientSecret'] },
+                            { required: ['useExistingClientSecret'] },
+                          ],
+                        },
+                      },
```

Comment:
I think it'll be nicer to define both authorizers indepenedently, and put them into `oneOf`.

For those few properties which are shared we may put definitions into variables, so maintainance wise code is not duplicated (and it doesn't serve as good case to create new definition)